src/OpenTelemetry.Instrumentation.Dependencies/HttpClientInstrumentationOptions.cs

```diff
@@ -54,54 +35,55 @@ internal HttpClientInstrumentationOptions(Func<string, object, object, bool> eve
         public ITextFormatActivity TextFormat { get; set; } = new TraceContextFormatActivity();
 
         /// <summary>
-        /// Gets or sets a hook to exclude calls based on domain or other per-request criterion.
+        /// Gets or sets an optional callback method for filtering HttpClient requests that are sent through the instrumentation.
         /// </summary>
-        internal Func<string, object, object, bool> EventFilter { get; set; }
+        public Func<HttpRequestMessage, bool> FilterFunc { get; set; }
 
-        private static bool DefaultFilter(string activityName, object arg1, object unused)
+        internal static bool IsInternalUrl(Uri requestUri)
         {
-            // TODO: there is some preliminary consensus that we should introduce 'terminal' spans or context.
-            // exporters should ensure they set it
+            var originalString = requestUri.OriginalString;
 
-            if (IsHttpOutgoingPostRequest(activityName, arg1, out Uri requestUri))
+            // zipkin
+            if (originalString.Contains(":9411/api/v2/spans"))
             {
-                var originalString = requestUri.OriginalString;
-
-                // zipkin
-                if (originalString.Contains(":9411/api/v2/spans"))
-                {
-                    return false;
-                }
+                return true;
+            }
 
-                // applicationinsights
-                if (originalString.StartsWith("https://dc.services.visualstudio") ||
-                    originalString.StartsWith("https://rt.services.visualstudio") ||
-                    originalString.StartsWith("https://dc.applicationinsights") ||
-                    originalString.StartsWith("https://live.applicationinsights") ||
-                    originalString.StartsWith("https://quickpulse.applicationinsights"))
-                {
-                    return false;
-                }
+            // applicationinsights
+            if (originalString.StartsWith("https://dc.services.visualstudio") ||
```

Comment:
I think we can remove this. Its leftover from old exporter

Reply:
Is the old exporter used at all or is it end-of-life? The reason I ask is if we remove this, and anyone uses it through NuGet or the contrib repo or wherever, we still need this to prevent an infinite export loop. We can only remove once we have a solve for the "terminal span" thing. I think? We could ask users to manually filter via the public API if they use the azure exporter?